### PR TITLE
feat(node): v15.3

### PR DESCRIPTION
**Pull request by SimonSchick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V15.md#15.1.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

cc @connor4312

### CODE REVIEW

File: types/node/diagnostic_channel.d.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+     */
+    function channel(name: string): Channel;
+
+    type ChannelListener = (name: string, message: unknown) => void;
+
+    /**
+     * Simple diagnostic channel that allows
+     */
+    class Channel {
+        readonly name: string;
```
Comment (connor4312) [inline types/node/diagnostic_channel.d.ts:25]: The channel has a [hasSubscribers] property too https://nodejs.org/docs/latest-v15.x/api/diagnostics_channel.html#diagnostics_channel_channel_hassubscribers

**Review by connor4312 — APPROVED** [review]

**Comment by SimonSchick** [comment]
Build failure unrelated.

**Comment by SimonSchick** [comment]
Added v15.3 as well, this will be the max version for the PR.

**Comment by victorperin** [comment]
@SimonSchick I have made a PR for version 16, can you look at it to see if there's any conflict with this?

I think if we work together we can update everything faster. 😃

**Comment by SimonSchick** [comment]
You can add my fork as an upstream and check yourself.
I'm hoping to get v15 typings up to 15.13 (v15.14 is security only) first, I already have another branch with all changes up to 15.6 ready.

The rationale here is: If we move to v16 straight away it's going to very annoying and time consuming to backport these changes as it will be difficult to reason which changes are already present and which are not.

The general expectation is that when you install version X that it contains all features of that version, while this is more of an upotic expectation, I think this is a reasonable expectation (and requirement) for definitions as fundamental as node.
Making a huge jump from v15.3 to 16.0 would leave a lot of gaps.

**Comment by SimonSchick** [comment]
@victorperin if you want you can prepare a PR that contains all changes explicitly listed in the v16 release notes and file it as a draft so we can iterate on it.

[I've worked up to 15.7 tonight](https://github.com/SimonSchick/DefinitelyTyped/tree/node/v15.9) and am optimistic to reach 15.14 by end of week.

**Comment by SimonSchick** [comment]
Build failure appears to be unrelated still.

**Comment by SimonSchick** [comment]
Rebased, again, hoping tests pass now, this should not block this PR though...

File: types/node/tsconfig.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
     ],
     "compilerOptions": {
         "module": "commonjs",
-        "target": "es6",
+        "target": "esnext",
```
Comment (andrewbranch) [inline types/node/tsconfig.json:8]: What’s this for?

Comment (SimonSchick) [reply types/node/tsconfig.json:8]: Been a moment but I believe this is to fix bigint support in tests.

**Review by andrewbranch — APPROVED** [review]

**Comment by SimonSchick** [comment]
Ready to merge

**Pull request merged**